### PR TITLE
If not armed, say "Wield foo" instead of "Dispose of none and wield foo"

### PR DESCRIPTION
**Pull request by sethsimon**
Fixes [#16855](https://github.com/CleverRaven/Cataclysm-DDA/issues/16855).

### CODE REVIEW

File: src/pickup.cpp [hunk 1]
```diff
@@ -278,9 +278,15 @@ pickup_answer handle_problematic_pickup( const item &it, bool &offered_swap, con
...
     offered_swap = true;
     // @todo Gray out if not enough hands
-    amenu.addentry( WIELD, !u.weapon.has_flag( "NO_UNWIELD" ), 'w',
+    if(u.is_armed()) {
+        amenu.addentry( WIELD, !u.weapon.has_flag( "NO_UNWIELD" ), 'w',
                     _("Dispose of %s and wield %s"), u.weapon.display_name().c_str(),
                     it.display_name().c_str() );
+    }
+    else {
+        amenu.addentry( WIELD, !u.weapon.has_flag( "NO_UNWIELD" ), 'w',
```
Comment (codemime) [inline src/pickup.cpp]: `!u.weapon.has_flag( "NO_UNWIELD" )` is a meaningless check if `u` is unarmed (always evaluates to `true`). You can simply write `true` there.

Comment (sethsimon) [inline src/pickup.cpp]: Good catch, thanks!

File: src/pickup.cpp [hunk 2]
```diff
@@ -278,9 +278,14 @@ pickup_answer handle_problematic_pickup( const item &it, bool &offered_swap, con
 
     offered_swap = true;
     // @todo Gray out if not enough hands
-    amenu.addentry( WIELD, !u.weapon.has_flag( "NO_UNWIELD" ), 'w',
+    if(u.is_armed()) {
+        amenu.addentry( WIELD, !u.weapon.has_flag( "NO_UNWIELD" ), 'w',
                     _("Dispose of %s and wield %s"), u.weapon.display_name().c_str(),
                     it.display_name().c_str() );
+    }
```
Comment (illi-kun) [inline src/pickup.cpp]: Pretty minor issue but anyway: in accordance with the project style, ```} else {``` should be in one line.

**Comment by sethsimon** [comment]
@illi-kun Good point. There were some other mis-styled lines in my changes, so I ran them through astyle.

**Pull request merged**